From 7ca2361487f0e6f8f56f941434e277f09130ef41 Mon Sep 17 00:00:00 2001
From: Mitsuhiro Kimura <mitsuhiro.kimura.kc@renesas.com>
Date: Fri, 29 Aug 2014 16:47:18 +0900
Subject: [PATCH] pinctrl: sh-pfc: r8a7794: Add EtherB pins to Ethernet pin
 groups

Signed-off-by: Mitsuhiro Kimura <mitsuhiro.kimura.kc@renesas.com>
---
 drivers/pinctrl/sh-pfc/pfc-r8a7794.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a7794.c b/drivers/pinctrl/sh-pfc/pfc-r8a7794.c
index f71fd17..ab85a46 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a7794.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a7794.c
@@ -3342,6 +3342,10 @@ static const char * const eth_groups[] = {
 	"eth_magic",
 	"eth_mdio",
 	"eth_rmii",
+	"eth_link_b",
+	"eth_magic_b",
+	"eth_mdio_b",
+	"eth_rmii_b",
 };
 
 static const char * const avb_groups[] = {
-- 
1.9.1

